### PR TITLE
lib/posix-sysinfo: Various improvements needed for OpenJDK support

### PR DESCRIPTION
**Pull request by csvancea**
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): [e.g. `x86_64` or N/A]
 - Platform(s): [e.g. `kvm`, `xen` or N/A]
 - Application(s): `app-elfloader`, OpenJDK support

### Description of changes

- Added a stub for `getcpu`. As discussed previously, it is hardcoded to return the first CPU (index 0).
- Revisited #678. Basically, glibc replaces calls to `sysconf(_SC_PHYS_PAGES)` with a syscall to `sysinfo`. The number of physical pages is then manually calculated by doing the morally equivalent of `totalram / PAGE_SIZE` [1]. Since `totalram` was being initialized to 0, `sysconf(_SC_PHYS_PAGES)` also returned 0, which lead to a failure in the Java heap initialization code.

With these changes, OpenJDK is finally able to boot up:

```
OpenJDK 64-Bit Server VM warning: sched_getaffinity failed (No such file or directory)- using online processor count (2) which may exceed available processors
OpenJDK 64-Bit Server VM warning: sched_getaffinity failed (No such file or directory)- using online processor count (2) which may exceed available processors
OpenJDK 64-Bit Server VM warning: sched_getaffinity failed (No such file or directory)- using online processor count (2) which may exceed available processors
OpenJDK 64-Bit Server VM warning: sched_getaffinity failed (No such file or directory)- using online processor count (2) which may exceed available processors
openjdk version "21-internal" 2023-09-19
OpenJDK Runtime Environment (slowdebug build 21-internal-adhoc.razvan.jdk)
OpenJDK 64-Bit Server VM (slowdebug build 21-internal-adhoc.razvan.jdk, mixed mode, sharing)
Console terminated, terminating guest (PID: 3914)...
```

[1] https://elixir.bootlin.com/glibc/glibc-2.35/source/sysdeps/unix/sysv/linux/getsysstats.c#L269

### CODE REVIEW

File: lib/posix-sysinfo/sysinfo.c [hunk 1]
```diff
@@ -72,16 +72,33 @@ static struct utsname utsname = {
...
+#ifdef CONFIG_HAVE_PAGING
+	pt = ukplat_pt_get_active();
+
+	total_memory = pt->fa->total_memory;
+	while (total_memory > __UL_MAX) {
+		total_memory >>= 1;
+		mem_unit <<= 1;
+	}
+
+	info->totalram = (unsigned long) (pt->fa->total_memory / mem_unit);
```
Comment (StefanJum) [inline lib/posix-sysinfo/sysinfo.c:97]: Isn't `pt->fa->total_memory / mem_unit` just `total_memory`?

**Review by StefanJum — COMMENTED** [review]
This looks fine @csvancea, I'll test it properly as soon as possible. Just one comment until then.

File: lib/posix-sysinfo/sysinfo.c (hunk 1 above)
Comment (csvancea) [reply lib/posix-sysinfo/sysinfo.c:97]: Sometimes. For x86_64 and arm64 (and basically all archs with sizeof(long) == 8) that is true, but you never know what other architectures we might support one day.

**Review by StefanJum — APPROVED** [review]
All good, thanks.

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request closed without merging**